### DISCUSSION TITLE
Override Exit method on App.xaml.cs

### DISCUSSION THREAD

**Discussion by rossirpaulo** [microsoft/WindowsAppSDK#3260] [Q&A]
How can we override the Exit method on WinUi3? I want to have a reliable way to perform some tasks before the program ends. Note that there is a way to override the OnLaunch method by:
```
    protected async override void OnLaunched(LaunchActivatedEventArgs args)
```

Is there something similar for closing/terminating the application?

**Comment by DarranRowe** [comment]
Main is overridable.
For C#, if you look for the generated App.g.i.cs file then you can work out how to do this. Make sure that it does everything that the currently defined main does.

**Comment by castorix** [comment]
It is usually done on [AppWindow.Closing](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.windowing.appwindow.closing?view=windows-app-sdk-1.2)

**Reply by rocksdanister** [reply]
It's closed in WinUI3 https://github.com/microsoft/microsoft-ui-xaml/issues/5454

**Reply by castorix** [reply]
> It's closed in WinUI3 [microsoft/microsoft-ui-xaml#5454](https://github.com/microsoft/microsoft-ui-xaml/issues/5454)

closed is for Microsoft.UI.Xaml.Window
and [cancel](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.windowing.appwindowclosingeventargs.cancel?view=windows-app-sdk-1.2#microsoft-ui-windowing-appwindowclosingeventargs-cancel) allows to intercept closing
